### PR TITLE
feat: add module for control-flow analysis

### PR DESCRIPTION
**Pull request by monperrus**
original code by @marcelinorc

### CODE REVIEW

**Comment by pvojtechovsky** [comment]
Cool! 🥇

**Comment by pvojtechovsky** [comment]
Looks like a nice Christmas present :-)

**Comment by surli** [comment]
@monperrus thanks for adding this one! Still I have a question regarding the license: the files you added do not have the license header. Does it mean they are in another license? What about the packaging of this module? What does it mean for the releases of Spoon? I really think we should be consistent on that subject: we already have a not-well-known license so let's avoid more ambiguities :)

**Comment by nharrand** [comment]
@surli I think in their [original repo](https://github.com/marcelinorc/spooncontrolflow) they had no header, and no licenses at all. As far as I understand, that means that we can do anything we want with them. 

So I agree we should put as header the license we want for spoon sub modules.

**Comment by surli** [comment]
> As far as I understand, that means that we can do anything we want with them.

We should have at least the agreement from @marcelinorc here and to put the original repository in the header then.

**Comment by nharrand** [comment]
Ok. I will write him an eamil then.

**Comment by monperrus** [comment]
> By default, the source code has the same copyright than books.
Interesting. Do you have an authoritative URL about this?

**Comment by nharrand** [comment]
Not necessarily authoritative: https://choosealicense.com/no-permission/

**Comment by surli** [comment]
https://en.wikipedia.org/wiki/License-free_software

> he software owner has not made explicit the terms of the license which makes the software fully copyright protected according to the Berne convention.

**Comment by monperrus** [comment]
https://en.wikipedia.org/wiki/Berne_Convention

**Comment by nharrand** [comment]
Marcelino has kindly accepted to put a MIT license on his project [as can be seen here](https://github.com/marcelinorc/spooncontrolflow/blob/master/README.md) so I think we can proceed with merging once we have updated the headers of these files.

**Comment by pvojtechovsky** [comment]
Thank you @marcelinorc !

**Comment by nharrand** [comment]
I think this should be ready for review.

File: spoon-control-flow/pom.xml [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+            <version>0.9.1</version>
+        </dependency>
+
+        <dependency>
+            <groupId>log4j</groupId>
+            <artifactId>log4j</artifactId>
+            <version>1.2.17</version>
+        </dependency>
+
+        <dependency>
```
Comment (surli) [inline spoon-control-flow/pom.xml]: junit, only for test no? And I thought @monperrus did a PR for bringing junit in the parent pom?

**Comment by surli** [comment]
Is it possible to add a Readme.md?

File: spoon-control-flow/pom.xml (hunk 1 above)
Comment (nharrand) [reply spoon-control-flow/pom.xml]: The PR in question is not merge-able yet #2828 if I am not mistaken.

Comment (nharrand) [reply spoon-control-flow/pom.xml]: But as soon as it is possible, I agree that we should move junit in parent.

**Comment by nharrand** [comment]
Yes definitely!

**Comment by monperrus** [comment]
Thanks a lot @nharrand for the last mile

**Comment by monperrus** [comment]
And kudos to @marcelinorc for the foundational work.

**Pull request merged**